core/workspace_svg.js

```diff
@@ -1369,8 +1369,23 @@ Blockly.WorkspaceSvg.prototype.markFocused = function() {
     Blockly.mainWorkspace = this;
     // We call e.preventDefault in many event handlers which means we
     // need to explicitly grab focus (e.g from a textarea) because
-    // the browser will not do it for us.
-    this.getParentSvg().focus();
+    // the browser will not do it for us.  How to do this is browser dependant
+    // as noted below.
+    
+    // Blur whatever was focused since explcitly grabbing focus below does not
+    // work in Edge.
+    if (document.activeElement) {
```

Comment:
Make all of this a separate function (setBrowserFocus()?) and keep the comments in there.  Then LGTM.